### DISCUSSION TITLE
为什么最多只能有2000个context，出于什么考虑？

### DISCUSSION THREAD

**Discussion by SenZh** [alibaba/Sentinel#2242] [Q&A]
为什么最多只能有2000个context，出于什么考虑？
如何突破这个限制？

**Comment by SomebodyHuang2017** [comment]
一个context对应一个EntranceNode，而EntranceNode会创建多个DefaultNode节点。node节点太多会对业务造成资源消耗。
所以需要对context做出限制。需要更深的理解可以看看context创建和NodeSelectorSlot的逻辑
